### PR TITLE
Updated cardano-api to 8.8, cardano-cli to 8.3.1.0 and cardano-node to 8.1.1

### PR DESCRIPTION
**Pull request by koslambrou**
* updated index-state of hackage and CHaP
* updated the iohk-nix flake dependency in order to have access to the new blst library when building core components
* removed cardano-testnet from marconi-chain-index because there is only an out-of-date version in CHaP. We will plan to replace it with cardano-node-emulator so it's okay.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense and have useful messages
    - [ ] Important changes are reflected in changelog.d of the affected packages
    - [ ] Relevant tickets are mentioned in commit messages
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Targeting main unless this is a cherry-pick backport
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] If relevant, reference the ADR in the PR and reference the PR in the ADR
    - [ ] Reviewer requested

### CODE REVIEW

File: nix/cells/marconi/library/make-marconi-project.nix [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            # See https://github.com/input-output-hk/cardano-node/issues/4194.
+            #
+            # The line 'export CARDANO_NODE_SRC=...' is used to specify the
+            # root folder used to fetch the `configuration.yaml` file (in
+            # marconi, it's currently in the
+            # `configuration/defaults/byron-mainnet` directory.
+            # Else, we'll get the error
+            # `/nix/store/ls0ky8x6zi3fkxrv7n4vs4x9czcqh1pb-marconi/marconi/test/configuration.yaml: openFile: does not exist (No such file or directory)`
+            marconi-chain-index.preCheck = "
+              export CARDANO_CLI=${cell.library.cardano-node.cardano-cli}/bin/cardano-cli${pkgs.stdenv.hostPlatform.extensions.executable}
```
Comment (koslambrou) [inline nix/cells/marconi/library/make-marconi-project.nix:94]: This also needs to change

File: marconi-chain-index/test-lib/Gen/Marconi/ChainIndex/Types.hs [hunk 2]
```diff
@@ -185,7 +185,20 @@ genTxBodyContentWithTxInsCollateral
...
+            initialPP
+              { C.protocolParamUTxOCostPerWord = Just 1
+              , C.protocolParamUTxOCostPerByte = Just 1
+              , C.protocolParamPrices = Just $ C.ExecutionUnitPrices 1 1
+              , C.protocolParamMaxTxExUnits = Just $ C.ExecutionUnits 1 1
+              , C.protocolParamMaxBlockExUnits = Just $ C.ExecutionUnits 1 1
+              , C.protocolParamMaxValueSize = Just 1
+              , C.protocolParamCollateralPercent = Just 1
+              , C.protocolParamMaxCollateralInputs = Just 1
+              }
```
Comment (berewt) [inline marconi-chain-index/test-lib/Gen/Marconi/ChainIndex/Types.hs:201]: It looks like these parameters are set in `cardano-api-gen:8.1.1.0` did you try to update it? https://github.com/input-output-hk/cardano-api/blob/cardano-api-gen-8.1.1.0/cardano-api/gen/Test/Gen/Cardano/Api/Typed.hs

File: marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/MintBurn.hs [hunk 3]
```diff
@@ -333,109 +310,106 @@ rewind = H.property $ do
       expected = filter isBeforeRollback events
   equalSet expected (MintBurn.fromRows queryResult)
 
+-- TODO Commented because cardano-testnet was out of date and we needed to update cardano-api.
```
Comment (berewt) [inline marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/MintBurn.hs:313]: Maybe add this comment as a note and move it to `marconi-chain-index/test-lib/Helpers.hs` and refers to it here?

**Review by berewt — APPROVED** [review]
LGTM.

File: marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/MintBurn.hs (hunk 3 above)
Comment (koslambrou) [reply marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/MintBurn.hs:313]: Good idea!

**Comment by james-iohk** [comment]
I would prefer the integration tests were not disable in case there is delay to using the emulator. At the least, it will ensure the tests remain up to date. It is possible to use cardano-testnet at the moment without Chaps. If we leave them disabled then, I hope we can get them working with the emulator soon.

File: marconi-chain-index/test-lib/Gen/Marconi/ChainIndex/Types.hs (hunk 2 above)
Comment (koslambrou) [reply marconi-chain-index/test-lib/Gen/Marconi/ChainIndex/Types.hs:201]: Actually, `cardano-api-gen-8.1.1.0` is empty. The link you provided is from cardano-api. So yeah, I'm using cardano-api-8.8 and these fields are still missing. Removing them makes the tests fail.

Comment (berewt) [reply marconi-chain-index/test-lib/Gen/Marconi/ChainIndex/Types.hs:201]: Oh, so we need to amend this once `cardano-api-gen` is updated then. :(

**Pull request merged**